code-gen-projects/java/code-gen-demo/src/test/java/org/example/CodeGenTest.java

```diff
@@ -3,24 +3,71 @@
 import org.junit.jupiter.api.Test;
 import static org.junit.jupiter.api.Assertions.*;
 import java.util.ArrayList;
+import com.amazon.ion.system.IonReaderBuilder;
+import com.amazon.ion.IonReader;
+import com.amazon.ion.system.IonTextWriterBuilder;
+import com.amazon.ion.IonWriter;
+import com.amazon.ion.IonSystem;
+import com.amazon.ion.system.IonSystemBuilder;
+import com.amazon.ion.IonLoader;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.io.File;
 
 class CodeGenTest {
-    // TODO: Add roundtrip tests after read-write APIs are supported for Java code generation
+    private static final IonSystem ionSystem = IonSystemBuilder.standard().build();
+    private static final IonLoader ionLoader = ionSystem.getLoader();
+
     @Test void getterAndSetterTestForStructWithFields() {
```

Comment:
FYI, the test says "getterAndSetterTest", but I don't see any use of setters.

Reply:
I will include tests for setter.